test/HostnameTest.php

```diff
@@ -506,6 +506,25 @@ public function testAdditionalUTF8TLDs()
         }
     }
 
+    public function testAdditionalPunycodedTLDs()
+    {
+        $validator = new Hostname(Hostname::ALLOW_ALL);
+        // Check UTF-8 TLD matching
+        $valuesExpected = [
```

Comment:
Please refactor using data providers.

Also split the test in two tests, one when is valid and another one when is not